packages/eservice-template-process/src/services/eserviceTemplateService.ts

```diff
@@ -2069,20 +2070,21 @@ async function updateDraftEServiceTemplateVersion(
     .exhaustive();
 
   const updatedAgreementApprovalPolicy = match(updateSeed)
-    .with({ type: "post" }, ({ seed }) =>
-      apiAgreementApprovalPolicyToAgreementApprovalPolicy(
-        seed.agreementApprovalPolicy
-      )
+    .with(
+      { type: "post" },
+      ({ seed }) =>
+        seed.agreementApprovalPolicy &&
+        apiAgreementApprovalPolicyToAgreementApprovalPolicy(
+          seed.agreementApprovalPolicy
+        )
```

Comment:
Better to be more explicit in the two cases using an if

```suggestion
      seed.agreementApprovalPolicy
        ? apiAgreementApprovalPolicyToAgreementApprovalPolicy(
            seed.agreementApprovalPolicy
          )
        : undefined
```